src/som/interpreter/actors/ResolvePromiseNode.java

```diff
@@ -23,19 +23,17 @@
    * if a promise resolution breakpoint was set.
    */
   @Specialization(guards = {"notAPromise(result)"})
-  public SResolver normalResolution(final VirtualFrame frame, final SResolver resolver, final Object result,
-      final boolean isBreakpointOnPromiseResolution) {
-    boolean breakpointOnResolution = isBreakpointOnPromiseResolution;
+  public SResolver normalResolution(final VirtualFrame frame,
+      final SResolver resolver, final Object result,
+      final boolean haltOnResolver, final boolean haltOnResolution) {
     SPromise promise = resolver.getPromise();
 
-    if (promise.isExplicitPromise() && promise.isTriggerExplicitPromiseResolverBreakpoint()) {
+    if (haltOnResolver || promise.haltOnResolver()) {
       haltNode.executeEvaluated(frame, result);
     }
-    if (promise.isExplicitPromise() && promise.isTriggerPromiseResolutionBreakpoint()) {
-      breakpointOnResolution = promise.isTriggerPromiseResolutionBreakpoint();
-    }
 
-    resolvePromise(Resolution.SUCCESSFUL, resolver, result, breakpointOnResolution);
+    resolvePromise(Resolution.SUCCESSFUL, resolver, result,
+        haltOnResolution || promise.haltOnResolution());
```

Comment:
As I understand when a promiseResolutionBreakpoint is set on a `createPromisePair` primitive for triggering the breakpoint, the implementation idea is the same: check a flag on the promise. With this new changes you reused the flag for triggering a promise resolution breakpoint, that you call `haltOnResolution`. I think it makes sense, I did not see that variant in that moment.